src/types.rs

```diff
@@ -54,11 +54,51 @@ impl From<&'static str> for Error {
 #[derive(Deserialize, Serialize)]
 pub struct AccountCredentials {
     pub(crate) id: String,
-    pub(crate) key_pkcs8: String,
+    /// Stored in DER, serialized as base64
+    #[serde(with = "pkcs8_serde")]
+    pub(crate) key_pkcs8: Vec<u8>,
     pub(crate) directory: Option<String>,
     pub(crate) urls: Option<DirectoryUrls>,
 }
 
+mod pkcs8_serde {
+    use std::fmt;
+
+    use base64::prelude::{Engine, BASE64_URL_SAFE_NO_PAD};
+    use serde::{de, Deserializer, Serializer};
+
+    pub(crate) fn serialize<S>(key_pkcs8: &[u8], serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let encoded = BASE64_URL_SAFE_NO_PAD.encode(key_pkcs8.as_ref());
```

Comment:
Isn't this basically doing PEM? https://www.rfc-editor.org/rfc/rfc7468#page-13
I think it makes sense to go full PEM and add the `BEGIN` and `END` parts here as well. This should allow more interop.

Reply:
I thought about it. (a) I don't really want to invest in writing a robust PEM parser, and I kind of don't want to add another dependency for it, (b) if we support PEM I worry that users will expect us to handle a bunch of other kinds of secret keys which PEM supports but which I'd prefer to avoid supporting.